File: agent/components/models/ticket.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class Ticket(BaseModel):
    id: int
    title: str
    description: str
    priority: str
    status: str
    customer_id: int
    customer_name: str
    tags: list[str] = []
    sla_due_at: Optional[datetime] = None
    created_at: datetime
    service_hint: Optional[str] = None
    ssh_host: Optional[str] = None
    ssh_user: Optional[str] = None
    ssh_port: int = 22

    def is_high_priority(self) -> bool:
        return self.priority.lower() in ("high", "critical")

    def is_open(self) -> bool:
        return self.status.upper() == "OPEN"

File: frontend/src/types/ticket.ts
export type Priority = 'high' | 'critical'
export type TicketStatus = 'OPEN' | 'DONE'

// Shape returned by GET /api/tickets/ (list) — matches ERP fields exactly
export interface Ticket {
  id: number
  title: string
  description: string
  priority: Priority
  status: TicketStatus
  customer_id: number
  customer_name: string
  sla_due_at: string | null
  created_at: string
  tags: string[]
  // Not returned by ERP list — only available after fetching detail
  service_hint?: string | null
  ssh_host?: string | null
  ssh_user?: string | null
  ssh_port?: number | null
}

// Shape returned by GET /api/tickets/{id} — ticket + flattened system info
export interface TicketDetail extends Ticket {
  ssh_host: string | null
  ssh_user: string | null
  ssh_port: number | null
  system_os?: string | null
  system_notes?: string | null
}
